### ISSUE TITLE
How does the overtraining detector work?

### ISSUE THREAD

**Issue by litsa-the-dancer** [IAHispano/Applio#365]
As of recently I've noticed you guys added an overtraining detector thingy. I'd like to know how it works and what graphs and values it takes into account.

**Comment by blaise-tk** [comment]
You can check this commit for more info https://github.com/IAHispano/Applio/commit/5593d183ed44d24639667ecb385fe14088b71693

In each epoch it stores the loss gen all of that point and from there it makes the comparisons.

**Comment by litsa-the-dancer** [comment]
So that's how it finds the global minimum? If so that really isnt all that accurate, its better for the model to look at loss/g/total rather than loss/gen/all. Looking at loss/g/total gives u a much better representation of how the generator is doing with the training data whilst also taking into account all the other factors such as mel, kl and fm.

**Comment by blaise-tk** [comment]
![image](https://github.com/IAHispano/Applio/assets/133521603/1fd12b28-1671-40db-8ee5-3e07e196f618)

**Comment by litsa-the-dancer** [comment]
I'll try testing it out
